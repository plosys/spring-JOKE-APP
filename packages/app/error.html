
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="initial-scale=1,minimum-scale=1,width=device-width" />
    <link rel="shortcut icon" href="/favicon.ico" />
    <link rel="icon" sizes="16x16" href="/icon-16.png" />
    <link rel="icon" sizes="32x32" href="/icon-32.png" />
    <link rel="icon" sizes="96x96" href="/icon-96.png" />
    <link rel="icon" sizes="128x128" href="/icon-128.png" />
    <link rel="icon" sizes="192x192" href="/icon-192.png" />
    <link rel="apple-touch-icon" sizes="76x76" href="/icon-76.png?maskable=true" />
    <link rel="apple-touch-icon" sizes="120x120" href="/icon-120.png?maskable=true" />
    <link rel="apple-touch-icon" sizes="152x152" href="/icon-152.png?maskable=true" />
    <link rel="apple-touch-icon" sizes="144x144" href="/icon-144.png?maskable=true" />
    <link rel="apple-touch-icon" sizes="152x152" href="/icon-152.png?maskable=true" />
    <link rel="apple-touch-icon" sizes="167x167" href="/icon-167.png?maskable=true" />
    <link rel="apple-touch-icon" sizes="180x180" href="/icon-180.png?maskable=true" />
    <!-- This *must* be the bottom one. -->
    <link rel="apple-touch-icon" href="/icon-57.png?maskable=true" />
    <title>Appsemble Â· App not found</title>
    <link rel="stylesheet" href="{{{ bulmaURL }}}" />
    <link rel="stylesheet" href="{{{ faURL }}}" />
  </head>
  <body>
    <div class="is-primary navbar">
      <header class="navbar-brand">
        <h1 class="navbar-item title is-4">Appsemble</h1>
      </header>
    </div>
    <section class="section">
      <p class="container">{{ message }}</p>
    </section>
  </body>
</html>