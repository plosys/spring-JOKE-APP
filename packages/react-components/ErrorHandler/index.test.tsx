import { render } from '@testing-library/react';
import { ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';

import { ErrorHandler } from './index.js';

it('should render its children if no errors are thrown', () => {
  function Child(): ReactElement {
    return <p>Test</p>;
  }
  function Fallback(): ReactElement {
    return <p>Something went wrong!</p>;
  }
  const { container } = render(
    <BrowserRouter>
      {/* eslint-disable-next-line react/jsx-no-bind */}
      <ErrorHandler fallback={Fallback}>
        <Child />
      </ErrorHandler>
    </BrowserRouter>,
  );

  expect(container).toMatchSnapshot();
});

it('should render its fallback when errors are thrown', () => {
  import.meta.jest.spyOn(console, 'error').mockImplementation();
  const Child = (): ReactElement => {
    throw new Error('test');
  };
  function Fallback(): ReactElement {
    return <p>Something went wrong!</p>;
  }
  const { container } = render(
    <BrowserRouter>
      {/* eslint-disable-next-line react/jsx-no-bind */}
      <ErrorHandler fallback={Fallback}>
        <Child />
      </ErrorHandler>
    </BrowserRouter>,
  );

  expect(container).toMatchSnapshot();
});
