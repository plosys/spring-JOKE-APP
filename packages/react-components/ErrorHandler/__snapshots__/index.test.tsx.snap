
// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render its children if no errors are thrown 1`] = `
<div>
  <p>
    Test
  </p>
</div>
`;

exports[`should render its fallback when errors are thrown 1`] = `
<div>
  <p>
    Something went wrong!
  </p>
</div>
`;